src/backends/perf_pt/mod.rs

```diff
@@ -223,15 +205,22 @@ impl Tracer for PerfPTTracer {
         Ok(())
     }
 
-    fn stop_tracing(&mut self) -> Result<(), TraceMeError> {
+    fn stop_tracing(&mut self) -> Result<HWTrace, TraceMeError> {
         let tr_ctx = self.tracer_ctx.ok_or(TraceMeError::TracerNotStarted)?;
+        let buf = ptr::null() as *const u8;
+        let len = 0;
         let rc = unsafe {
-            perf_pt_stop_tracer(tr_ctx)
+            perf_pt_stop_tracer(tr_ctx, &buf, &len)
         };
         if rc == -1 {
             return Err(TraceMeError::CFailure);
         }
-        Ok(())
+        assert!(buf != ptr::null()); // The C code should have put the buffer address here.
```

Comment:
I'm not sure this `assert` is useful. Does the API says that it might not set `buf`? If not, we should just assume that things work (if the ptr is null, things will go badly wrong soon enough) without getting overly-defensive.

Reply:
ok

Reply:
The only case where ptr is set to null, is in failure, which should be caught by the previous line.

Reply:
OK, we can kill this line then.